### DISCUSSION TITLE
Nested class with Construct

### DISCUSSION THREAD

**Discussion by Trikieu123** [pydantic/pydantic#4806] [Question]
Hi. I have a list of trusted  `dict()` data and ideally want to skip the validation during the serialization. 

```
class B(BaseModel):
    _b: int = 0

class A(BaseModel):
    _a: int = 0
    temp_b: B = None

def test_mock():

    b = {'b': 2}
    _dict = {'a': 2, 'temp_b': b}
    a = A.construct(**_dict) 
```

Realizing that the construct did not initiate the `temp_b` as a instance of the nested class B. 
```
type(a.temp_b)
<class 'dict'>
```

As been documented, the `construct` should behave as the `__int__` but this seems to be not.

**Comment by anthonycorletti** [comment]
👋 Hi @Trikieu123, after checking out [these docs](https://docs.pydantic.dev/usage/models/#creating-models-without-validation) it looks like there is no validation applied to an `A.construct` so it makes sense that you don't see `temp_b` as a validated instance of `B`.

This example helped me see that the returned `A` objects here are very different when `.construct` is used and not used:

```Python
from pydantic import BaseModel


class B(BaseModel):
    b: int


class A(BaseModel):
    a: int
    b: B


data = {
    "a": 2,
    "b": {
        "b": 2,
    },
}


a = A(**data)
print("a:\n", a)
print("a.dict():\n", a.dict())

a_construct = A.construct(values=data)
print("a_construct:\n", a_construct)
print("a_construct.dict():\n", a_construct.dict())


assert type(a) == type(a_construct)

assert type(a) == A
assert type(a.a) == int
assert type(a.b) == B

assert type(a_construct) == A
assert not hasattr(a_construct, "a")
assert not hasattr(a_construct, "b")

# a:
#  a=2 b=B(b=2)
# a.dict():
#  {'a': 2, 'b': {'b': 2}}
# a_construct:
#  values={'a': 2, 'b': {'b': 2}}
# a_construct.dict():
#  {'values': {'a': 2, 'b': {'b': 2}}}
```

An approach that might work for you could be having some classes that you use for validation and others that you use for quick serialization post-validation.

I hope this helps!
